Applications/Spire/Source/SpireTester/CachedOrderImbalanceIndicatorModelTester.cpp

```diff
@@ -0,0 +1,317 @@
+#include <catch.hpp>
+#include "Spire/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.hpp"
+#include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+#include "Spire/Spire/QtPromise.hpp"
+#include "Spire/SpireTester/TestOrderImbalanceIndicatorModel.hpp"
+#include "Spire/SpireTester/SpireTester.hpp"
+
+using namespace boost;
+using namespace boost::icl;
+using namespace boost::posix_time;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  auto make_imbalance(const std::string& symbol, const ptime& timestamp) {
+    return OrderImbalance(Security(symbol, 0), Side::BID, 100,
+      Money(Money::ONE), timestamp);
+  }
+
+  const auto A = make_imbalance("A", from_time_t(100));
+  const auto B = make_imbalance("B", from_time_t(200));
+  const auto C = make_imbalance("C", from_time_t(300));
+  const auto D = make_imbalance("D", from_time_t(400));
+  const auto E = make_imbalance("E", from_time_t(500));
```

Comment:
You need a test case where multiple imbalances happen at the same time and you will need to ensure that they all got loaded properly.

Reply:
Can you elaborate on this, I don't really understand it.

Reply:
Sure, add a test case roughly as follows:

* Publish an imbalance with timestamp X.
* Load all imbalances within range [X - 1s, X + 1s]
* Expect the list [X]
* Publish an imbalance with timestamp X.
* Load all imbalances within range [X - 1s, X + 1s]
* Expect the list [X, X]

Reply:
I thought you said earlier that it wouldn't happen in practice?

Reply:
Then I misspoke, almost all imbalances are published at the exact same time of the day, namely at 3:40 PM. To avoid miscommunications keep questions on Asana because I might mishear the question when spoken orally or I might misspeak or trip on my words. If it's written then I'm a lot less likely to make such a mistake.